### DISCUSSION TITLE
Unification of estimator structure (EKF) issue clarification

### DISCUSSION THREAD

**Discussion by siddarth-murugesan** [bitcraze/discussions#450] [Q&A]
External project - "A Modified EKF for Constant Altitude Flights" 
(https://www.bitcraze.io/2022/01/a-modified-ekf-for-constant-altitude-flights - https://github.com/bitcraze/crazyflie-firmware/pull/823)

There was a reason stated by bitcraze that the unification of the estimator structure (EKF) was needed in this project, to merge it to the master of bitcraze firmware. As I am also doing a similar project with the crazy flie, I was thinking of fixing it if I could know some more information on the requirements for the unification of the estimator structure (EKF). Are the two newly added states (f,r) in EKF changes and their increased computations the issue here?

Thank you
Siddarth

**Comment by knmcguire** [comment]
@dominiknatter, could you help out here perhaps?
